src/test/java/uk/gov/pay/directdebit/mandate/services/MandateServiceTest.java

```diff
@@ -315,4 +318,49 @@ public void shouldThrowWhenSubmittingTheSameDetailsSeveralTimes_andCreateOnlyOne
         assertThat(newMandate.getState(), is(SUBMITTED));
         verify(mockedDirectDebitEventService, times(1)).registerDirectDebitConfirmedEventFor(newMandate);
     }
+
+    @Test
+    public void shouldSetMandateStatusToExpired_FromCreated() {
+        Mandate mandate = MandateFixture.aMandateFixture()
+                .withState(MandateState.CREATED)
+                .withCreatedDate(ZonedDateTime.now().minusMinutes(91L))
+                .toEntity();
+        service.mandateExpiredFor(mandate);
+        verify(mockedDirectDebitEventService).registerMandateExpiredEventFor(mandate);
+        verify(mockedMandateDao).updateState(mandate.getId(), MandateState.EXPIRED);
+        assertThat(mandate.getState(), Is.is(MandateState.EXPIRED));
+    }
+
+    @Test
+    public void shouldSetMandateStatusToExpired_FromSubmitted() {
+        Mandate mandate = MandateFixture.aMandateFixture()
+                .withState(MandateState.SUBMITTED)
+                .withCreatedDate(ZonedDateTime.now().minusMinutes(91L))
+                .toEntity();
+        service.mandateExpiredFor(mandate);
+        verify(mockedDirectDebitEventService).registerMandateExpiredEventFor(mandate);
+        verify(mockedMandateDao).updateState(mandate.getId(), MandateState.EXPIRED);
+        assertThat(mandate.getState(), Is.is(MandateState.EXPIRED));
+    }
+
+    @Test
+    public void shouldSetMandateStatusToExpired_FromDdDetails() {
+        Mandate mandate = MandateFixture.aMandateFixture()
+                .withState(MandateState.AWAITING_DIRECT_DEBIT_DETAILS)
+                .withCreatedDate(ZonedDateTime.now().minusMinutes(91L))
+                .toEntity();
+        service.mandateExpiredFor(mandate);
+        verify(mockedDirectDebitEventService).registerMandateExpiredEventFor(mandate);
+        verify(mockedMandateDao).updateState(mandate.getId(), MandateState.EXPIRED);
+        assertThat(mandate.getState(), Is.is(MandateState.EXPIRED));
+    }
```

Comment:
the above 3 tests look like a good candidate for parameterized tests. i know TestNG does it but it might be too big a change to migrate from junit4 completely. https://github.com/Pragmatists/JUnitParams might work. check with @mrlumbu, but I'd be happy for this to be in another PR so I can progress this PR by merging it.

Reply:
Yes we definitely don't want to migrate off junit. Personally I don't think parameterised tests sound lie a particularly great idea - I would rather have very verbose, very explicit tests. Happy to discuss that at some point though 😄

Reply:
hmm ok... let's agree to disagree then ;)
just one more pedantic thing... can we static import the `Is.is` in the tests?